docs/source/pipelines.rst

```diff
@@ -4,194 +4,321 @@
+information about importing and running a Databricks notebook, see
+https://docs.databricks.com/notebooks/notebooks-manage.html#import-a-notebook.
+
+You can also modify the example to build and score a model on a dataset of your choosing.
+For more information about modifying the |MLflow Pipelines Regression Template|, see the
+|Regression Template reference guide|.
+
+Key concepts
 ------------
 
 .. _steps-key-concept:
 
-Steps
-~~~~~
-
-Steps represent individual modeling or MLOps procedures, such as ingesting data,
-fitting an estimator, evaluating a model against test data, or deploying a model for
-real-time scoring. Steps accept a collection of well-defined inputs and produce
-well-defined outputs according to user-defined configurations and code. MLflow Pipelines
-intelligently caches step outputs, ensuring that steps are only executed if their
-inputs, code, or configurations have changed, or if such changes have occurred in dependent steps.
-Steps are composed to form :ref:`Pipelines <pipelines-key-concept>`.
+- **Steps**: Steps represent individual ML operations, such as ingesting data, fitting an estimator,
+  evaluating a model against test data, or deploying a model for real-time scoring. Steps accept a
+  collection of well-defined inputs and produce well-defined outputs according to user-defined
+  configurations and code.
 
-.. code-block:: python
-  :caption: Example Python code defining an estimator that is fit during the |train step|
-            of the :ref:`MLflow Regression Pipeline <mlflow-regression-pipeline>`
+.. _pipelines-key-concept:
 
-  def estimator_fn():
-    from sklearn.linear_model import SGDRegressor
+- **Pipelines**: A Pipeline is a composition of :ref:`Steps <steps-key-concept>` used to solve an
+  ML problem or perform an MLOps task, such as developing a regression model or performing batch
+  model scoring on production data. MLflow Pipelines provides
+  :py:class:`APIs <mlflow.pipelines.Pipeline>` and a :ref:`CLI <cli>` for running pipelines and
+  inspecting their results.
 
-    return SGDRegressor(random_state=42)
+.. _pipeline-templates-key-concept:
 
-.. code-block:: yaml
-  :caption: Example YAML configuration defining train / validation / test dataset split
-            ratios for the |split step| of the :ref:`MLflow Regression Pipeline
-            <mlflow-regression-pipeline>`.  This configuration demonstrates use of
-            |Jinja2| templating syntax to define a dynamic value for the ``split_ratios``
-            attribute that can be overriden by
-            :ref:`Pipeline Profiles <profiles-key-concept>`.
-
-  steps:
-    split:
-      # Train/validation/test split ratios
-      split_ratios: {{SPLIT_RATIOS|default([0.75, 0.125, 0.125])}}
+- **Templates**: A Pipeline Template is a git repository with a :ref:`standardized, modular layout
+  <pipeline-template-structure>` containing all of the customizable code and configurations for a
```

Comment:
```suggestion
  <pipeline-template-structure>` containing all of the customizable code and configurations required for a
```

Reply:
I don't think the word `required` is necessary here.

Reply:
I feel like something is missing gramatically, if you feel `required` is too prescriptive then maybe `code and configurations used by a ...`  or something?